### ISSUE TITLE
Missing validation: Can't transition (or otherwise use) swapchain images before acquiring them

### ISSUE THREAD

**Issue by hrydgard** [KhronosGroup/Vulkan-ValidationLayers#365]
Most operations on swapchain images are illegal before acquiring them, including barrier-transitioning their layouts: "An application can acquire use of a presentable image from the presentation engine. Use of a presentable image must occur only after the image is returned by vkAcquireNextImageKHR, and before it is presented by vkQueuePresentKHR. This includes transitioning the image layout and rendering commands."

We accidentally did just this, resulting in https://github.com/hrydgard/ppsspp/issues/11417  - random crashes on a specific device. A hard to track down bug, heroically solved by Hans-Kristian.

The validation layers should definitely warn about this.

**Comment by karl-lunarg** [comment]
I agree it isn't an enhancement.  I'll remove the tag and @jzulauf-lunarg can put it back with an explanation if it still needs that tag.  This does have the incomplete tag, which is accurate and would not let the enhancement tag influence the triage and priority of the issue.

**Comment by spencer-lunarg** [comment]
update:

- partial fix in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3382
- new issue about it https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/3426
- "fixed" https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3890
- ANGLE found issue https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/3915
  - was reverted https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/3928
- Attempted to fix again but PR got stale - https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/4004

This should also get a proper VUID as well
